thin-replica-server/test/thin_replica_server_test.cpp

```diff
@@ -574,7 +574,9 @@ class TestStateMachine {
         if ((current_event_group_to_send_ + storage_.total_egs_for_other_clients == (last_event_group_to_send_ - 1)) &&
             not more_event_groups_to_add_) {
           on_finished_dropping_event_groups();
-        } else if (live_buffer_->Full() || live_buffer_->oldestEventGroupId() > (storage_.getLastEventGroupId() + 1)) {
+        } else if (live_buffer_->Full() ||
+                   (!(live_buffer_->EmptyEventGroupQueue()) &&
```

Comment:
Thanks for adding this check. I think we should also add a similar check i.e., `live_buffer_->Size() > 0` for legacy events [here](https://github.com/vmware/concord-bft/blob/88cb7a23191cd082351c209c133c442f3909e840/thin-replica-server/test/thin_replica_server_test.cpp#L554). -
`else if (live_buffer_->Full() || live_buffer_->Size() > 0 && live_buffer_->oldestBlockId() > (storage_.getLastBlockId() + 1))`

Reply:
I have added this check for the legacy events case in recent revisions to this PR.